### ISSUE TITLE
GPX.writeFeatures mess up ele and time

### ISSUE THREAD

**Issue by sookoll** [openlayers/openlayers#9996]
**Describe the bug**
When writing features to GPX, time property (time: "2003-05-25T00:00:00") become <ele>1053820800</ele> node in output gpx and ele property are ignored. Time node in gpx will be: <time>NaN-NaN-NaNTNaN:NaN:NaNZ</time>

**To Reproduce**
```
import GPX from 'ol/format/GPX'
// validate, that time and ele property exists
console.log(fset[0].get('time'), fset[0].get('ele'))
const string = new GPX().writeFeatures(fset)
console.log(string)
```

**Expected behavior**
Well, ele property should match to ele node and time property to time node on wp, rtept, trkpt

**Comment by simonseyock** [comment]
Thanks for reporting and the example!
I tried the example with this gpx file: https://openlayers.org/en/latest/examples/data/gpx/fells_loop.gpx
It exported the date correctly:
```
<?xml version="1.0" encoding="utf-8"?><gpx xmlns="http://www.topografix.com/GPX/1/1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd" version="1.1" creator="OpenLayers"><wpt lat="42.44729799999999" lon="-71.121447"><ele>127.7112</ele><time>2001-06-02T03:26:58Z</time><name>5156</name><desc>5156</desc><sym>Dot</sym><type>Intersection</type></wpt></gpx>
```
I checked the imported features and the `ele` and `time` properties are actually stored as the third and forth elements of the coordinate array of the geometry. I don't know if this is the most intuitive behaviour and why it is implemented this way. I think it might be better to write and read this values from the feature.

**Comment by simonseyock** [comment]
@tschaub, @ahocevar we probably would accept a pull request here?

**Comment by simonseyock** [comment]
Is this maybe because waypoints could have different times but would not all be converted to features but to one LineString?

**Comment by simonseyock** [comment]
I think the least that could be done here is to improve the documentation. And that time is written as `NaN-NaN-NaNTNaN:NaN:NaNZ` if not provided should be fixed.

**Comment by simonseyock** [comment]
Maybe at least using the current time there would be nicer - or omitting the `<time>` node if gpx allows for that.

**Comment by sookoll** [comment]
If waypoint do not contain ele tag, ol write time as third element in coordinates array. This is bug for sure. For example geocaching.com exported gpx do not contain elevation. Waypoints are always converted to Points. Route and Tracks are converted to Linestrings.

**Comment by simonseyock** [comment]
Yes you are right: time and ele are optional (See https://www.topografix.com/gpx/1/1/#type_wptType)
So this would need to get handled correctly.

**Comment by sookoll** [comment]
Can ol handle coordinates in form [x, y, null, null]? Would it be solution here? It is true, that storing elevation and time as feature attributes works only for waypoints.

**Comment by simonseyock** [comment]
i put together a little fiddle to play around with GPX:
https://codesandbox.io/s/gpx-cs4bo

I used `undefined`, it actually seem to work quite ok. 

If any element is not set it will not get written. Can you provide a the geometry of a feature so it actually writes `NaN-NaN-NaNTNaN:NaN:NaNZ` to the time node?
